nixos/modules/services/desktops/geoclue2.nix

```diff
@@ -61,6 +186,104 @@ in
         wantedBy = [ "default.target" ];
       };
     };
+
+    environment.etc."geoclue/geoclue.conf".text = ''
+    # Configuration file for Geoclue
+
+    # Agent configuration options
+    [agent]
+
+    # Whitelist of desktop IDs (without .desktop part) of all agents we recognise,
+    # separated by a ';'.
+    whitelist=${let whitelist = optional cfg.enableDemoAgent "geoclue-demo-agent" ++ singleton "gnome-shell" ++ cfg.extraWhitelistedAgents; in semiColon whitelist}
+
+    # Network NMEA source configuration options
+    [network-nmea]
+
+    # Fetch location from NMEA sources on local network?
+    enable=${tf cfg.enableNmea}
+
+    # WiFi source configuration options
+    [wifi]
+
+    # URL to the wifi geolocation service. The key can currenty be anything, just
+    # needs to be present but that is likely going to change in future.
+    url=${cfg.geoProviderUrl}
+
+    # Submit data to Mozilla Location Service
+    # If set to true, geoclue will automatically submit network data to Mozilla
+    # each time it gets a GPS lock.
+    #
+    submit-data=${tf cfg.submitData}
+
+    # URL to submission API of Mozilla Location Service
+    submission-url=${cfg.submissionUrl}
+
+    # A nickname to submit network data with. This is currently used for leaderboard:
+    # https://location.services.mozilla.com/leaders
+    # A nickname must be 2-32 characters long.
+    submission-nick=${cfg.submissionNick}
+
+    # Application configuration options
+    #
+    # NOTE: Having an entry here for an application with allowed=true means that
+    #       geoclue will not ask agent to authorize the application. This is to
+    #       ensure that system components and applications with built-in
+    #       authorization mechanism (e.g web browsers) do not have to be bound to
+    #       agents.
+    #
+    #       If your application is denied access to location information and your
+    #       operating system doesn't provide any mechanism to change that, it is
+    #       likely a bug in your operation system (or geoclue). The solution is to
+    #       report the issue with all details, rather than adding your application
+    #       to this list.
+    #
+    # Format:
+    #
+    # # Desktop ID of application without .desktop part
+    # [random-app]
+    #
+    # # Allowed access to location information?
+    # allowed=true|false
+    #
+    # # Is application a system component?
+    # system=true|false
+    #
+    # # List of UIDs of all users for which this application is allowed location
+    # # info access, separate by ';'. Keep it empty for allowing it for all users.
+    # users=
+
+    [gnome-datetime-panel]
+    allowed=true
+    system=true
+    users=
+
+    [gnome-color-panel]
+    allowed=true
+    system=true
+    users=
+
+    [org.gnome.Shell]
```

Comment:
Could we not move these to the gnome module now?

Reply:
These are the defaults upstream. Though I feel the use of geoclue2 implies gnome does it not?

Reply:
Nope, geoclue is an independent freedesktop component. Also we do not have gnome-flashback module yet, so gnome-panel sections would have nowhere to go.